src/test/java/org/folio/rest/api/HoldingsSourceTest.java

```diff
@@ -0,0 +1,308 @@
+
+    assertThat(sources.size(), is(2));
+  }
+
+  @Test
+  public void cannotReplaceANonexistentHoldingsSource() throws MalformedURLException, InterruptedException, ExecutionException, TimeoutException {
+	  Response sourceResponse = holdingsSourceClient.attemptToReplace(UUID.randomUUID().toString(), new JsonObject()
+		      .put("name", "updated source name"));
+	  assertThat(sourceResponse.getStatusCode(), is(HttpURLConnection.HTTP_NOT_FOUND));
+  }
+
+  @Test
+  public void canRemoveAHoldingsSource() throws MalformedURLException, InterruptedException, ExecutionException, TimeoutException {
+	  IndividualResource holdingsSource = holdingsSourceClient.create(
+			  new JsonObject()
+			  	.put("name", "deleteable source")
+			  );
+	  UUID deleteableHoldingsSourceId = holdingsSource.getId();
+	  holdingsSourceClient.delete(deleteableHoldingsSourceId);
+
+	  Response deleteResponse = holdingsSourceClient.getById(deleteableHoldingsSourceId);
+	  assertThat(deleteResponse.getStatusCode(), is(HttpURLConnection.HTTP_NOT_FOUND));
+
+  }
+
+  @Test
+  public void canNotRemoveSpecialHoldingsSources() throws MalformedURLException, InterruptedException, ExecutionException, TimeoutException {
+      IndividualResource holdingsSource = holdingsSourceClient.create(
+              new JsonObject()
+                .put("name", "source with folio source")
+                .put("source", "folio")
+              );
+      IndividualResource holdingsSource2 = holdingsSourceClient.create(
+          new JsonObject()
+            .put("name", "source with marc source")
+            .put("source", "marc")
+          );
+      UUID folioHoldingsSourceId = holdingsSource.getId();
+      UUID marcHoldingsSourceId = holdingsSource2.getId();
+
+      Response folioDeleteResponse = holdingsSourceClient.attemptToDelete(folioHoldingsSourceId);
+      Response marcDeleteResponse = holdingsSourceClient.attemptToDelete(marcHoldingsSourceId);
+
+      //they should not have been deleted:
+      assertThat(folioDeleteResponse.getStatusCode(), is(HttpURLConnection.HTTP_BAD_REQUEST));
+      assertThat(marcDeleteResponse.getStatusCode(), is(HttpURLConnection.HTTP_BAD_REQUEST));
+  }
+
+  @Test
+  public void canNotRemoveHoldingsSourcesAttachedToHoldings() throws MalformedURLException, InterruptedException, ExecutionException, TimeoutException {
+    UUID instanceId = UUID.randomUUID();
+    UUID sourceId = UUID.randomUUID();
+
+    JsonObject instanceToCreate = new JsonObject();
+
+    instanceToCreate.put("id", instanceId.toString());
+    instanceToCreate.put("source", "Test Source");
+    instanceToCreate.put("title", "Test Instance");
+    instanceToCreate.put("instanceTypeId", "535e3160-763a-42f9-b0c0-d8ed7df6e2a2");
+
+    instancesClient.create(instanceToCreate);
+
+    holdingsSourceClient.create(
+              new JsonObject()
+              .put("id", sourceId.toString())
+              .put("name", "associated source")
+            ).getJson();
```

Comment:
Why is the indentation and formatting of this is different to the other similar examples in this class?

Reply:
I activated the .editorconfig from the git repo and this is the indentation it wants to do.

Reply:
`# default to 2 space indentation
[*]
trim_trailing_whitespace = true
insert_final_newline = true
max_line_length = 80
indent_style = space
indent_size = 2`

https://github.com/folio-org/mod-inventory-storage/blob/master/.editorconfig

Reply:
Hmm, that is very strange, because most of the indentation in this file is using four space indentation as far as I can tell (or double indentation). 

I decided not to raise it, as it was consistent.